libs/gl-client/src/lnurl/tor_http_client.rs

```diff
@@ -0,0 +1,48 @@
+use anyhow::Result;
+use arti_client::{TorClient, TorClientConfig};
+use log::info;
+use tokio::io::{AsyncReadExt, AsyncWriteExt};
+use url::Url;
+
+pub struct TorHttpClient {
+    tor_client: TorClient<tor_rtcompat::PreferredRuntime>,
+}
+
+impl TorHttpClient {
+    pub async fn new() -> Result<Self> {
+        let mut config_builder = TorClientConfig::builder();
+        let address_filter = config_builder.address_filter();
+        address_filter.allow_onion_addrs(true);
+        address_filter.build()?;
+        config_builder.build()?;
+
+        let tor_client = TorClient::create_bootstrapped(config_builder.build().unwrap()).await?;
+        Ok(Self { tor_client })
+    }
+
+    pub async fn tor_http_get(&mut self, url: Url) -> Result<()> {
+        info!("Connecting to onion service. This may take a while...");
+        let mut stream = self
+            .tor_client
+            .connect((url.domain().unwrap(), 80))
+            .await
+            .unwrap();
+
+        info!("Connecting to {}...", url.to_string());
+        let http_message = format!(
+            "GET {} HTTP/1.1\r\nHost: {}\r\n\r\n",
+            url.path(),
+            url.domain().unwrap()
+        )
+        .as_bytes()
+        .to_owned();
```

Comment:
Does `reqwest` not allow setting up a separate `Connector` that is Tor-aware? Doing an open-coded HTTP implementation here is probably out of scope (missing content-type handling, chunking, multipart return, TLS support, ...).

Reply:
The goal here was not to make a full http client. There's no need for the use case of resolving lnurls. Maybe naming it `TorHttpClient` gave the wrong impression of the intended goal. 

All the examples of connecting a reqwest http client over Tor uses proxies. The arti crate [supports running a socks proxy](https://docs.rs/arti/latest/arti/#arti) but there were less examples for that so I opted to go for the higher-level arti-client crate which showed the client making [manual http requests in the docs](https://docs.rs/arti-client/0.10.0/arti_client/#example-making-connections-over-tor).

I'll defer the discussion to a later PR where I'll re-include the now-removed `tor_http_client`.